### PR TITLE
Fix tile duplication

### PR DESCRIPTION
**Pull request by makson96**
For some reason it seems that game name in gameinfo file can be different that the one obtained from API. As a result minigalaxy identifies those titles as separate games:
stellaris_duplication
Identifying games by ID seems to be more reliable idea. In this pull request I want to address this issue by changing game recognition (for tile creating) from game titles to game ids.
I know this is very regression prone part of the code, so I am also introducing unit tests for it to make sure that any future modifications won't break it.

### CODE REVIEW

**Comment by sharkwouter** [comment]
Yeah, I have observed this as well, good catch. Does this code also work for games which do not have an id on disk?

**Comment by makson96** [comment]
Well... it does not. I will try to prepare valid unit test for such a situation tomorrow and address it in the code.

**Comment by makson96** [comment]
The issue you mentioned is now addressed and I have also added unit test for checking such a situations.

**Review by sharkwouter — CHANGES_REQUESTED** [review]
This looks pretty good, but there is one thing I noticed which was a bit strange. I made a comment on that. Thanks for the work on this.

File: minigalaxy/ui/library.py [hunk 1]
```diff
@@ -137,19 +139,19 @@ def __get_installed_games(self) -> List[Game]:
...
-            GLib.idle_add(self.parent.show_error, _("Failed to retrieve library"), _("Couldn't connect to GOG servers"))
-            return
-        installed_game_names = []
-        for game in self.games:
-            installed_game_names.append(game.name.lower())
-        for game in retrieved_games:
-            if game.name.lower() not in installed_game_names:
-                self.games.append(game)
+
+def add_games_from_api(self_games, self_api):
```
[the remark this answers is not in the document]
Comment (makson96) [reply minigalaxy/ui/library.py:143]: Well, I am unable to initialize this class for the sake of unit tests...

**Pull request merged**